### DISCUSSION TITLE
node ace command return 0 even the argument is not match

### DISCUSSION THREAD

**Discussion by bstsnail** [adonisjs/core#1541] [Help]
![image](https://user-images.githubusercontent.com/16055826/91692909-c887f400-eb9c-11ea-94cb-1d832ca2fd49.png)
How can I get the real exit code of node ace command ?

**Comment by RodolfoSilva** [comment]
Reported here: https://github.com/adonisjs/ace/issues/119

**Comment by RodolfoSilva** [comment]
@bstsnail, @thetutlage I've made this workaround:

Create a new file, in the same place of `ace` file, and name it like: `ace-ci`.

Put the script below, inside the `ace-ci` file:

```js
#!/usr/bin/env node
const cp = require('child_process')

let ace = cp.spawn('node', ['ace', ...process.argv.slice(2)], {
  stdio: [process.stdin, process.stdout, 'pipe'],
})

let hasError = false

ace.stderr.on('data', (buffer) => {
  if (!hasError) {
    hasError = /\[[^\]]*error[^\]]*\]\s+/gi.test(buffer.toString())
  }
  process.stderr.write(buffer)
})

ace.on('close', () => {
  if (hasError) process.exit(1)
})
```

And now, instead use the regular `ace`, just use `ace-ci`.

**Note:** you still need the `ace` file, this script only check if the output has some error message, and fire exit code 1 in the end.

![image](https://user-images.githubusercontent.com/1703020/97210882-fc4a5900-179c-11eb-8277-2f68625047d8.png)

![image](https://user-images.githubusercontent.com/1703020/97210874-f81e3b80-179c-11eb-8ad1-39ae07095cef.png)

**Comment by thetutlage** [comment]
It is a bug. Will work on the fix
